File: gitdot-api/src/endpoint/repository/list_repository_commit_filters.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{RepositoryCommitFilterResource, common::Page},
};

pub struct ListRepositoryCommitFilters;

impl Endpoint for ListRepositoryCommitFilters {
    const PATH: &'static str = "/repository/{owner}/{repo}/commit_filters";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListRepositoryCommitFiltersRequest;
    type Response = ListRepositoryCommitFiltersResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListRepositoryCommitFiltersRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListRepositoryCommitFiltersResponse = Page<RepositoryCommitFilterResource>;

File: gitdot-web/packages/api/src/endpoint/repository/list-repository-commit-filters.ts
import { z } from "zod";
import { page, RepositoryCommitFilterResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListRepositoryCommitFiltersRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListRepositoryCommitFiltersRequest = z.infer<
  typeof ListRepositoryCommitFiltersRequest
>;

export const ListRepositoryCommitFiltersResponse = page(
  RepositoryCommitFilterResource,
);
export type ListRepositoryCommitFiltersResponse = z.infer<
  typeof ListRepositoryCommitFiltersResponse
>;

export const ListRepositoryCommitFilters = {
  path: "/repository/{owner}/{repo}/commit-filters",
  method: "GET",
  request: ListRepositoryCommitFiltersRequest,
  response: ListRepositoryCommitFiltersResponse,
} as const satisfies Endpoint;
export type ListRepositoryCommitFilters = typeof ListRepositoryCommitFilters;
